### PR TITLE
Add threat model json schema

### PR DESCRIPTION
**Pull request by jgadsden**
**Summary**
It would be good to provide a JSON schema that threat models could be validated against

**Description for the changelog**
add json schema for threat model

**Other info**

### CODE REVIEW

**Comment by jgadsden** [comment]
Hello @pdxjohnny, here is the completed json schema. Is this looking OK? I have put it all in one 400 line file ... hope that is normal thing to do

File: docs/development/schema/owasp.threat-dragon.schema.json [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+        "reviewer": {
+          "description": "The reviewer of the overall threat model",
+          "type": "string"
+        }
+      },
+      "required": [ "contributors", "diagrams" ]
+    }
+  },
+  "required": [ "summary", "detail" ]
+}
```
Comment (pdxjohnny) [inline docs/development/schema/owasp.threat-dragon.schema.json:407]: Will validate today

**Review by pdxjohnny — COMMENTED** [review]
Thank you! Will run some code to confirm, but this looks great to me so far as I can see.

File: docs/development/schema/owasp.threat-dragon.schema.json (hunk 1 above)
Comment (jgadsden) [reply docs/development/schema/owasp.threat-dragon.schema.json:407]: Note that this is the version 1.x schema, and we are hoping that version 2.0 Threat Dragon (which is still in development) will keep to this schema
At present a version 2.0 threat model will fail the schema - but we are working on that

Comment (jgadsden) [reply docs/development/schema/owasp.threat-dragon.schema.json:407]: and I have corrected an error where the 'tdVersion' was in the wrong place

**Comment by jgadsden** [comment]
Merging this in to the main branch because more detail changes are being added from other branches

**Pull request merged**